### DISCUSSION TITLE
Using multiple test client users for multiple devices/browsers testing in Parallel

### DISCUSSION THREAD

**Discussion by RyanRedpath** [webdriverio/webdriverio#6815] [Q&A] [answered]
Hi, 

I wanted to reach out to see if anyone had any advice or information with regards to using multiple devices, each using a unique device login user with test login credentials? We have a long list of test account credentials, but unsure how best to utilise this effectively.  

The main issue we have is our app uses single session, so if I am running some tests on say 10 mobile devices in BrowserStack at the same time in parallel, I will be logging out other other devices as the tests run. So essentially we want to use a different login user for each device or browsers (or up to 10 as that’s all we can run in parallel). Can/should we set this in the capabilities somehow? Or maybe using WDIO config?

We’ve tried a few different ways of trying to achieve this but nothing we are really happy  with(or that’s working for us) so I wondered if you guys could offer any help or guidance with our investigation as I am sure WDIO must offer an easy way of doing this that we are missing!?

**Comment by christian-bromann** [comment, marked as answer]
> Can/should we set this in the capabilities somehow? Or maybe using WDIO config?

Yes, within the capabilities you can always set your custom configs, e.g.:

```js
// wdio.conf.js
exports.config = {
    capabilities: [{
         browserName: 'chrome',
         'custom:options': {
             username: 'foo',
             password: 'bar'
         }
    }, {
         ...
    }],
    // ...
}
```

and then access it via:

```js
browser.requestedCapabilities['custom:options'].username
```

Can you check if this works for you?

**Reply by RyanRedpath** [reply]
Thanks @christian-bromann , i thought that 'custom:options' looked like a winner, I still do in fact but currently I get an error about W3C WebDriver capabilities.

`[0-0] 2021-05-05T14:24:01.610Z ERROR @wdio/runner: Error: Invalid or unsupported WebDriver capabilities found ("project", "build", "app", "browserstack.idleTimeout", "chromedriverExecutableDir", "name", "device", "os_version"). Ensure to only use valid W3C WebDriver capabilities (see https://w3c.github.io/webdriver/#capabilities).`

**Reply by christian-bromann** [reply]
Please make sure that you use proper W3C capabilities, check BS platform configurator

**Reply by RyanRedpath** [reply]
Yes indeed, just strange that it didn't show this message before adding the 'custom:options', we shall work it out, thanks for the help.
